### ISSUE TITLE
Concatenating files without a newline

### ISSUE THREAD

**Issue by cpaika** [sbt/sbt-assembly#489]
Using `sbt-assembly` 2.0.0 I've seen an issue on file concatenation - when one file does not have a newline at the end of it (in our case a HOCON `reference.conf`) the concatenated next file will not be concatenated with a newline, resulting in the contents being incorrect, example:

File1 (no newline):
```
akka.property = on
```

File2:
```
next-file-config{
  #more configuration
```

Result:
```
akka.property = onnext-file-config{
  #more configuration
```

Downgrading to `sbt-assembly` 1.2.0 fixed this issue so it feels like a regression since then.

**Comment by eed3si9n** [comment]
@cpaika Thanks for the report!

**Comment by dragisak** [comment]
#482 is the fix for this

**Comment by eed3si9n** [comment]
O wow. Thanks for the reminder!

**Comment by eed3si9n** [comment]
The fix PR is merged, and released as 2.1.0.
